Bmp ribout

lsang6WIND opened the pull request:
Integrate https://github.com/FRRouting/frr/pull/14847/
    add label to adj out
    do not activate addpath by default in bmp
    add new topotests: adj out vpn prefix, addpath bmp tests

ton31337 commented:
Why did you close github.com/FRRouting/frr/pull/18719 instead of addressing the comments? It's hard to track what's were changed or not now...

lsang6WIND commented:
ci:rerun unrelated

mxyns commented on tests/topotests/lib/bmp_collector/bgp/update/nlri.py:
```diff
@@ -47,21 +47,113 @@ class NlriIPv4Unicast:
-        prefix = padding(data[1:], 4)
 
-        return {"ip_prefix": f"{ipaddress.IPv4Address(prefix)}/{prefix_len}"}
+        # we have an add-path id, this check is simpler than the unnecessary (i think?)
```
I think here we still need the `detect_addpath_prefix_ipv46` for the case where the NLRI is: addpath_id + (pfx_len=0)
that would be 5 bytes with an addpath id and according to [RFC4271](https://datatracker.ietf.org/doc/html/rfc4271) is apparently valid:
```
 a) Length:

    The Length field indicates the length in bits of the IP
    address prefix.  A length of zero indicates a prefix that
    matches all IP addresses (with prefix, itself, of zero
    octets).

 b) Prefix:

    The Prefix field contains an IP address prefix, followed by
    the minimum number of trailing bits needed to make the end
    of the field fall on an octet boundary.  Note that the value
    of trailing bits is irrelevant.
```

I think `detect_addpath_prefix_ipv46` also catches this edge case

lsang6WIND replied:
I have some additional adjustments for addpath that are not included in this PR; I have now pushed them.

mxyns replied:
Looks like the scapy script is (or is based on) the same one Wireshark snippet I had based `detect_addpath_prefix_ipv46 ` so that looks good to me :)

mxyns reviewed (commented):
Thank you @lsang6WIND for taking over PR #14847

Quick question, what did you fix for "fix adj out withdraw message" ? I might have missed the changes for this one

Everything else seems fine to me

Thanks for fixing everything and nice solution for the addpath capable in bmp_monitor, I was not aware that the first allocated ID was 1 and not zero :)

lsang6WIND commented:
thank you for taking the time to review

> Quick question, what did you fix for "fix adj out withdraw message" ? I might have missed the changes for this one
 
I have dropped that patch and log is now updated.

pguibert6WIND commented:
ci:rerun

mxyns commented:
> I have dropped that patch and log is now updated.
Okay, out of curiosity, what was that patch trying to fix?

mwinter-osr commented:
CI:rerun Rerunning after aborting due to CI upgrade

mwinter-osr commented:
CI:rerun Another rerun test

donaldsharp commented:
I'm confused about why we have moved the PR to a new one.  `I accidently closed it` doesn't hold water with me.  Accidently re-open it then.
